pkg/controller/update_cluster.go

```diff
@@ -595,10 +611,11 @@ func (c *M3DBController) findPodInstanceToRemove(
 		if err != nil {
 			return nil, nil, pkgerrors.WithMessage(err, "error finding pod in placement")
 		}
-		return pod, inst, nil
+		leftToRemove--
+		podsToRemove = append(podsToRemove, pod)
+		instancesToRemove = append(instancesToRemove, inst)
 	}
-
-	return nil, nil, errNoPodsInPlacement
+	return podsToRemove, instancesToRemove, nil
```

Comment:
To ensure this is doing what the user intends, do we want to return an error if `len(podsToRemove) != removeCount`? Not sure if there's a case where that could happen.

Reply:
not sure, podsToRemove seems to only be used in tests for assertions. We ignore it in production code.

Reply:
Also I don't think that can happen with current impl, since both arrays are only appended under same conditions.

Reply:
I think the only case where this can happen is if we somehow want to remove more than exists. There is a test case when removeCount is for 4 but we have only 3 instances in placement.

In practice this should not happen because we remove only if placement contains more than desired:
```
		if inPlacement > desired {
			setLogger.Info("remove instance from placement for set")
			return c.shrinkPlacementForSet(cluster, set, placement, int(inPlacement-desired))
		}
```

. Maybe if we spam expand/shrink we could enter some weird state here. But I think it's hard to reason what will happen anyways. 

Another approach that potentially can be safer is to instead of having "removeCount" we could pass in "desired" so that we try to reach target state, we could than assert if target state makes sense like "desired" > "currentPodCount" otherwise it looks like expansion.